controllers/searchController.js

```diff
@@ -0,0 +1,131 @@
+import { Op } from 'sequelize';
+import ServerResponse from '../modules';
+import models from '../database/models';
+
+const { User, Article, Tag } = models;
+const { successResponse, errorResponse } = ServerResponse;
+
+/**
+ *
+ *
+ * @export
+ * @class ArticleController
+ */
+class SearchController {
+/**
+ * perform custom search sitewide
+ *
+ * @param {Object} req Express request
+ * @param {Object} res Express object
+ * @param {Object} next Express object
+ * @returns {Array} articles, authors and tags
+ */
+  static async customSearch(req, res, next) {
+    const { keyword } = req.query;
+
+    if (!keyword) {
+      return errorResponse(res, 400, 'Please input a search parameter');
+    }
+    try {
+      const articleAttributes = [
+        'title',
+        'slug',
+        'body',
+        'description',
+        'status',
+        'readTime',
+        'createdAt'
+      ];
+
+      const userAttributes = ['firstName', 'lastName'];
+
+      const tagAttributes = ['name'];
+
+      const articles = await Article.findAll({
+        attributes: articleAttributes,
+        where: {
+          title: {
+            [Op.iLike]: `%${keyword}%`,
+          }
+        },
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: ['firstName', 'lastName'],
+          }
+        ]
+      });
+
+      const authors = await User.findAll({
+        attributes: userAttributes,
+        where: {
+          [Op.or]: [
+            {
+              firstName: {
+                [Op.iLike]: `%${keyword}%`,
+              }
+            },
+            {
+              lastName: {
+                [Op.iLike]: `%${keyword}%`
+              }
+            }
+          ]
+        },
+        include: [
+          {
+            model: Article,
+            as: 'articles',
+            attributes: ['slug', 'title', 'description', 'body', 'readTime'],
+            include: [
+              {
+              // tag
+                model: Tag,
+                as: 'tags',
```

Comment:
Hey Emeka, isn't there a way to search all the related tables and columns in one query? You could do this by combining the sequelize operators

Reply:
thanks, done that!